test/unit/tempfile_spec.lua

```diff
@@ -1,58 +1,64 @@
-local lfs = require 'lfs'
-local helpers = require 'test.unit.helpers'
+local lfs = require('lfs')
+local helpers = require('test.unit.helpers')(after_each)
+local itp = helpers.gen_itp(it)
 
-local os = helpers.cimport './src/nvim/os/os.h'
-local tempfile = helpers.cimport './src/nvim/fileio.h'
+local eq = helpers.eq
+local neq = helpers.neq
+local cimport = helpers.cimport
+local deferred_call = helpers.deferred_call
+local separate_cleanup = helpers.separate_cleanup
+
+local lib = cimport('./src/nvim/os/os.h', './src/nvim/fileio.h')
 
 describe('tempfile related functions', function()
-  before_each(function()
-    tempfile.vim_deltempdir()
-  end)
-  after_each(function()
-    tempfile.vim_deltempdir()
+  before_each(deferred_call(function()
+    lib.vim_deltempdir()
+  end))
+  separate_cleanup(function()
```

Comment:
What is `separate_cleanup`? How is it different from `teardown` and `after_each`? A brief comment attached to `separate_cleanup` would help. Probably `deferred_call` also.